### PR TITLE
{FLINK-6965] Include snappy-java in flink-dist

### PR DESCRIPTION
**Pull request by zentol**
This PR adds removes the snappy dependency exclusion from flink-core.

Without this dependency Flink components that work with avro, and thus potentially snappy, fail when loading snappy. This also happens if snappy is provided in the user-jar since the flink component doesn't use the user class loader.

To prevent this dependency from being removed again in the future by accident i modified `checkShadedArtifacts()` function in the travis scripts to check for the presence of the dependency.

### CODE REVIEW

File: flink-core/pom.xml [hunk 1]
```diff
@@ -79,13 +79,12 @@ under the License.
...
-			<!-- managed version -->
-			<exclusions>
-				<exclusion>
-					<groupId>org.xerial.snappy</groupId>
-					<artifactId>snappy-java</artifactId>
-				</exclusion>
-			</exclusions>
+		</dependency>
+
+		<!-- We explicitly depend on snappy since connectors that require it load it through the system class loader -->
```
Comment (tzulitai) [inline flink-core/pom.xml]: Side question: is a fix also required for connectors to correctly use the user classloader in such cases?

Comment (zentol) [inline flink-core/pom.xml]: for the avro/snappy case, no. This PR should work as is.

File: tools/travis_mvn_watchdog.sh [hunk 2]
```diff
@@ -182,6 +182,13 @@ check_shaded_artifacts() {
 		exit 1
 	fi
 
+	SNAPPY=`cat allClasses | grep '^org/xerial/snappy' | wc -l`
+	if [ AVRO == "0" ]; then
```
Comment (packet23) [inline tools/travis_mvn_watchdog.sh]: Should it not read
```bash
if [ $SNAPPY == "0" ]; then
   # bail out
fi
```

Comment (zentol) [inline tools/travis_mvn_watchdog.sh]: yes! nice catch...

**Comment by StephanEwen** [comment]
+1

Please merge this for `master` and `release-1.3`.

**Pull request closed without merging**